subsys/bluetooth/host/l2cap.c

```diff
@@ -1194,7 +1194,7 @@ static void le_ecred_conn_req(struct bt_l2cap *l2cap, uint8_t ident,
 
 response:
 	buf = l2cap_create_le_sig_pdu(buf, BT_L2CAP_ECRED_CONN_RSP, ident,
-				      sizeof(*rsp) + (sizeof(scid) * i));
+				      sizeof(*rsp) + (sizeof(scid)));
```

Comment:
This doesn't look right to me, as `scid` is just an 16-bit value. So this will not, contrary to the commit message, respond with the length containing the full size of the `scid` array. 

I guess what you need to do is to calculate how many channels were requested based on the length field, and use that value instead of `i`.

Reply:
Of course, you are right: test didn't caught that, as it was requesting just one channel. I updated the PR, now requested CID count is based on `buf` size, after `req` is extracted.